src/components/qr-scanner/qr-scanner.ts

```diff
@@ -41,6 +41,21 @@ export class QRScannerComponent {
 
     if (qrCode.match(prefixUriRegex)) {
       scheme.address = prefixUriRegex.exec(qrCode)[1];
+      const paramsRegex = new RegExp(`(\\?|\\&)([^=]+)\\=([^&]+)`, 'g');
+      let regexResult;
+      while ((regexResult = paramsRegex.exec(qrCode)) != null) {
```

Comment:
I'm not sure how this loop can ever change? `qrCode` never changes, which would surely result in an infinite loop if `regexResult != null`?

Reply:
paramsRegex is updated each time exec is runned.

"If the match succeeds, the exec() method returns an array and updates properties of the regular expression object. The returned array has the matched text as the first item, and then one item for each capturing parenthesis that matched containing the text that was captured."

source : https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec

Reply:
Any news / need for some rework ?

Reply:
@alexbarnsley